### ISSUE TITLE
Demos no longer functioning?

### ISSUE THREAD

**Issue by gantproductions** [vimeo/vimeo-depth-player#17]
Hello all! I've been very interested in exploring Depthkit + Vimeo for my projects but it seems none of the demos are functioning (volumetric playback and volumetric live-streaming). Is there a way to fix these? I've tried them on Safari, Firefox, and Chrome.

**Comment by tangbao** [comment]
Vimeo just deleted these two videos so the player can no longer request the video from their server successfully. Also, some APIs in WebVR are not available in Chrome 80, I tried 73 and that could work.

By the way, I am interested in what kind of project you are working on, could you please share something about it?

**Comment by tangbao** [comment]
> Hello all! I've been very interested in exploring Depthkit + Vimeo for my projects but it seems none of the demos are functioning (volumetric playback and volumetric live-streaming). Is there a way to fix these? I've tried them on Safari, Firefox, and Chrome.

Hi, I found that vimeo recovered the video. For now I can test the demo successfully on Chrome 74.

**Comment by oliverellmers** [comment]
Has anybody managed to get the demos working locally yet?

I have cloned the repository, installed all dependencies, added a VIMEO_TOKEN, and updated the link to my uploaded depth video recording on Vimeo.

The node server reports no errors, that the Environment variables are successfully loaded from the .env file, running on port 53... and that a GET request was made to /

Is there anything else I am missing to be able to run the examples? Has anybody had any success?

**Comment by tangbao** [comment]
> Has anybody managed to get the demos working locally yet?
> 
> I have cloned the repository, installed all dependencies, added a VIMEO_TOKEN, and updated the link to my uploaded depth video recording on Vimeo.
> 
> The node server reports no errors, that the Environment variables are successfully loaded from the .env file, running on port 53... and that a GET request was made to /
> 
> Is there anything else I am missing to be able to run the examples? Has anybody had any success?

Yes mine works locally. Please use Chrome 73 or 74 (the only two versions I tested).

**Comment by oliverellmers** [comment]
@tangbao thanks for the quick reply

Unfortunately, this still does not resolve the issue and it is still just sitting loading.

Have I missed any important steps or is there anything you can think of I may not be doing correctly?

I have followed the instructions according to the readme and am receiving no errors

**Comment by tangbao** [comment]
@oliverellmers Have you opened the debug tool (in Chrome, press F12, then jump to Console) and see if there is any error?

**Comment by oliverellmers** [comment]
Okay I have found the problem (I won't say error as unsure this is an error) at line 76 in demo.html

`if (WEBVR.hasWebVR()) {
        renderer.vr.enabled = true;
        document.body.appendChild(WEBVR.createButton(renderer));
      }
      else {
        controls = new THREE.OrbitControls(camera);
        controls.target.set(0, 0.9, -1.0);
        controls.update();
      }`
It seems as though the browser is assuming a device is attached, and then hangs while trying to execute `document.body.appendChild(WEBVR.createButton(renderer));`

I resolved it by bypassing this all together and simply setting the controls to use the `THREE.OrbitControls()` and it is now working

**Comment by tangbao** [comment]
Congrats!

**Comment by oliverellmers** [comment]
Haha thanks!

I think this is still potentially a bug though - not sure what the expected behaviour is meant to be at that point...

I am also interested in what @gantproductions mentioned at the beginning of this thread about **volumetric live-streaming** - is this implying you can live stream from an azure at home to vimeo via this toolset? If so, is there an example or demo?

thanks in advance!!

**Comment by tangbao** [comment]
@oliverellmers That's almost the same thing as the local one. Just deploy your depth player on the cloud, then visit the player using http://cloud_ip:port_number. Make sure you configure your firewall correctly.

**Comment by gantproductions** [comment]
Hello @tangbao and @oliverellmers . I began a project awhile back where I was researching volumetric live streaming solutions. Due to many factors including the pandemic, cancellations, time frame, and of course budget, this project was abandoned. However, I did become quite knowledgeable on what similar solutions already exist, and then began exploring other realtime vfx programs that incorporate sensors and streaming. Feel free to reach out to me if you want to geek out and chat. In the meantime, I still very much stay extremely interesting in anything related to Depthkit and a live streaming solution, so fingers crossed!
